blog/BLOG_part2.md

```diff
@@ -0,0 +1,152 @@
+
+# Making an animated gif from a recorded Nethack-game - Part 2 / 2
+
+## Part 2: Making an animated gif with PHP
+I came across ttyrec file, a file that contains recorded Nethack-game. NetHack is an open source single-player roguelike video game, first released in 1987. I wanted to make an animated gif out of it.
+
+![](images/NethackScreenshot.gif)
+[Image Source: Wikipedia](https://commons.wikimedia.org/wiki/File:NethackScreenshot.gif)
+
+[In part one we have interpretted a ttyrec-file into a 6415 different gifs.](BLOG.md)
+
+I have a transformed a ttyrec-file into a 6415 different gifs. How to make an animated gif out of those gifs? Here again there is ready tools starting from ffmpeg, a command tool that is capable of doing any kind of media files and transformations. Then again, I want to define the delays directly in code so that I can get the delays exactly the same as in original ttyrec-file and possibly modify the delays. I started with some googling. What are the options? Has someone already made this? 
+
+My limitations were that I wanted the tool to be done with PHP as the vt100-terminal interpretter was already done with that and I could package them to together. All I could find was a piece of code that was done 12 years ago. The code was also copied across the internet but the code looked pretty bad. It had no comments, the variables were non descriptive and there were very clear errors in it. After a while decided that I had no other options. Copied the code and started thinking what was happening there. After some debugging, I realized that the code builds the animated gif from existing gifs to a new gif with bitwise operations and at the end concatenating individual gifs into a single gif.
```

Comment:
So there weren't actually any errors in the code, it was just a first hunch?

Reply:
Ah. true. Syntax errors were there.  Make this a bit lighter.

Reply:
It was not completely clear since first there "were very clear errors in it" and then starts the talk about how the code was ingenious but really hard to decipher. Just a small notion but was a bit unsure about that. Text was fairly easy to read, even thought i haven't been using PHP much